### ISSUE TITLE
SE v4.0.3 is experiencing a lot of repetition when using Whisper large(v3) for voice recognition.

### ISSUE THREAD

**Issue by msjsc001** [SubtitleEdit/subtitleedit#7991]
![image](https://github.com/SubtitleEdit/subtitleedit/assets/93178788/6f986375-a385-414d-a8cd-791e70855b72)

**Comment by darnn** [comment]
I'm fairly certain that's a CPP cuBLAS issue, not a Subtitle Edit issue. You can check by downloading it separately and running it on the file through the command line. Though I will say I've encountered it too. Your best bet in terms of a quick fix is probably `Tools`->`Merge lines with same text`, but that wouldn't help in this case, since it's two repeating lines.
You can also try to mess around with adding command line arguments (if you click `Advanced` in the Whisper window you have in your screenshot), but I have yet to try this myself, so I'm not sure what would resolve it. Maybe adding an initial prompt (something like `--prompt "Hello, today we'll be talking about swing trade."`) might help.

**Comment by msjsc001** [comment]
> I'm fairly certain that's a CPP cuBLAS issue, not a Subtitle Edit issue. You can check by downloading it separately and running it on the file through the command line. Though I will say I've encountered it too. Your best bet in terms of a quick fix is probably `Tools`->`Merge lines with same text`, but that wouldn't help in this case, since it's two repeating lines. 重试    错误原因 You can also try to mess around with adding command line arguments (if you click `Advanced` in the Whisper window you have in your screenshot), but I have yet to try this myself, so I'm not sure what would resolve it. Maybe adding an initial prompt (something like `--prompt "Hello, today we'll be talking about swing trade."`) might help. 重试    错误原因

I switched to using large(v2), and everything works fine. The difference between the two models isn't significant, so I'll stick with large(v2). I'm not sure what the issue with large(v3) is.

**Comment by darnn** [comment]
That's odd! But I can definitely tell you it's the way CPP cuBLAS is interacting with large-v3, because I use large-v3 with WhisperX (through Kaggle) all the time (plus sometimes vanilla Whisper and Whisper-timestamped), and it never does this, but with CPP cuBLAS I've seen it happen.

**Comment by msjsc001** [comment]
> 真奇怪！但我可以肯定地告诉你，这是 CPP cuBLAS 与 large-v3 交互的方式造成的，因为我一直将 large-v3 与 WhisperX（通过 Kaggle）一起使用（有时也会使用 vanilla Whisper 和 Whisper-timestamped），它从未出现过这种情况，但我看到 CPP cuBLAS 出现了这种情况。

Thank you for sharing.

**Comment by fraternl** [comment]
I never used the "cuBlas" setting and I have had repetitive lines in several subtitles.
I'm talking about the single repetitive lines, though

**Comment by darnn** [comment]
Oh yes, this is something that does happen with Whisper in general (though it's improved a lot as time has gone on). But the way it normally happens is you'll have, for example, one sentence repeating for ten minutes straight, instead of whatever else was supposed to be there. Whereas with this CPP cuBLAS thing, it's something like this: In the actual audio, the person is saying: "I had the thought of staging a show, not about the place, but about the myth of the place."
And then the subtitles it would produce would be:

`13`
`00:01:43,940 --> 00:01:53,260`
`I had the thought of staging a show, not about the place, but about the myth of the place`
` `
`14`
`00:01:53,260 --> 00:01:53,260`
`I haging a show, not about the place, but about the myth of the place`
` `
`15`
`00:01:53,260 --> 00:01:53,260`
`I haging a show, not about the place, but about the myth of the place`
` `
`16`
`00:01:53,260 --> 00:01:53,260`
`I haging a show, not about the place, but about the myth of the place`

So, the first subtitle is mostly correct, and then there are three more subtitles that are a less correct version of the first, they're identical, and their duration is zero. Which is not how it happens with other versions of Whisper, at least not in my experience. I only started seeing it when I started using CPP cuBLAS (the latest version, with large-v3).
